src/mon/OSDMonitor.cc

```diff
@@ -8648,6 +8661,39 @@ int OSDMonitor::prepare_command_pool_set(const cmdmap_t& cmdmap,
 	ss << "pg_autoscale_bias must be between 0 and 1000";
 	return -EINVAL;
       }
+    } else if (var == "dedup_tier") {
+      int64_t lowtierpool_id = osdmap.lookup_pg_pool_name(val);
+      if (lowtierpool_id < 0) {
+	ss << "unrecognized pool '" << val << "'";
+	return -ENOENT;
+      }
+      const pg_pool_t *tp = osdmap.get_pg_pool(lowtierpool_id);
+      ceph_assert(tp);
+      if (p.is_erasure()) {
+	ss << "pool '" << poolstr
+	   << "' is an ec pool, which cannot be a tier";
+	return -ENOTSUP;
+      }
+      n = lowtierpool_id;
+      interr.clear(); 
```

Comment:
I think you need to actually check interr here.  If I'm understanding the code correctly, val is only valid if interr.empty(), otherwise interr indicates why val couldn't be parsed.

Reply:
I think I'm not following what happens here.  Rereading it, it looks like interr should not be relevant here as the param is actually a string, so it should always be empty?  You've added a check that returns an error if interr is empty.  I don't understand why it would normally be non-empty.

Reply:
Hm.. Input val is string (e.g., pool_name), so interr is true and interr.empty() is false here. But, the actual value supposed to store is pool_opts_t::INT(pool_name --> pool_id). As a result, if I don't clear interr, -EINVAL will be returned.